packages/caravan-fees/src/cpfp.ts

```diff
@@ -0,0 +1,105 @@
+import { Transaction } from "bitcoinjs-lib-v5";
+import { unsignedMultisigTransaction } from "@caravan/bitcoin";
+import BigNumber from "bignumber.js";
+import { CPFPOptions, UTXO } from "./types";
+import { estimateVirtualSize } from "./utils";
+
+/**
+ * Prepare a CPFP (Child-Pays-for-Parent) transaction.
+ * This function doesn't broadcast the transaction, it just prepares it.
+ * @param options CPFP options including the parent transaction and new fee rate
+ * @returns A new transaction ready for signing and broadcasting
+ */
+export function prepareCPFPTransaction(options: CPFPOptions): Transaction {
+  const {
+    parentTransaction,
+    newFeeRate,
+    availableUTXOs,
+    destinationAddress,
+    network,
+    multisigDetails,
+  } = options;
+
+  // Find suitable output from parent transaction to spend
+  const parentOutput = findSuitableParentOutput(
+    parentTransaction,
+    availableUTXOs
+  );
+  if (!parentOutput) {
+    throw new Error("No suitable output found in parent transaction for CPFP");
+  }
+
+  const childSize = estimateChildTransactionSize({
+    inputCount: 1, // We're spending one output from the parent
+    outputCount: 2, // Destination output and potentially change
+    addressType: multisigDetails.addressType,
+    requiredSigners: multisigDetails.requiredSigners,
+    totalSigners: multisigDetails.totalSigners,
```

Comment:
nit: I think you could just do this. 
```suggestion
    ...multisigDetails
```

Reply:
I guess now we are not using this , though options for cpfp and rbf classes do have requiredSigners , totalSigners which are not accessed via the spread operator because we need to handle cases like if the user does not provide dust threshold then we set it to default one